catroid/src/main/java/org/catrobat/catroid/io/asynctask/ProjectRenameTask.java

```diff
@@ -54,21 +54,31 @@ public ProjectRenameTask setListener(ProjectRenameListener listener) {
 		return this;
 	}
 
-	public static boolean task(File projectDir, String dstName) {
+	public static File task(File projectDir, String dstName) throws IOException {
 		File dstDir = new File(projectDir.getParent(), FileMetaDataExtractor.encodeSpecialCharsForFileSystem(dstName));
 
 		try {
-			return projectDir.renameTo(dstDir)
-					&& XstreamSerializer.renameProject(new File(dstDir, CODE_XML_FILE_NAME), dstName);
+			if (projectDir.renameTo(dstDir)
+					&& XstreamSerializer.renameProject(new File(dstDir, CODE_XML_FILE_NAME), dstName)) {
+				return dstDir;
+			} else {
+				return null;
```

Comment:
still not loving null returns and the null checks that are needed then. :crying_cat_face: 

how about throwing a different exception here? e.g. `throw new IOException("some cool message")`